src/app/services/theme/theme.service.ts

```diff
@@ -0,0 +1,202 @@
+import { DOCUMENT } from "@angular/common";
+import { Inject, Injectable } from "@angular/core";
+import { IS_SERVER_PLATFORM } from "src/app/app.helper";
+
+export const bawThemes = [
+  "highlight",
+  "primary",
+  "secondary",
+  "success",
+  "info",
+  "warning",
+  "danger",
+  "light",
+  "dark",
+] as const;
+
+export const bawThemeVariants = [
+  "",
+  "-lighter",
+  "-lightest",
+  "-darker",
+  "-darkest",
+] as const;
+
+export type BawTheme = typeof bawThemes[number];
+export type BawThemeVariant = typeof bawThemeVariants[number];
+
+export interface RGB {
+  red: number;
+  green: number;
+  blue: number;
+}
+
+export interface HSL {
+  hue: number;
+  saturation: number;
+  lightness: number;
+}
+
+@Injectable({
+  providedIn: "root",
+})
+export class ThemeService {
+  public constructor(
+    @Inject(DOCUMENT) private document: Document,
+    @Inject(IS_SERVER_PLATFORM) private isServer: boolean
+  ) {}
+
+  /**
+   * Set the base colour for a theme
+   *
+   * @param theme Theme to modify
+   * @param color New base colour for theme
+   */
+  public setTheme(theme: BawTheme, color: string): void {
```

Comment:
This should be named something more like _setColorVariable_. A _setTheme_ function should update all the colours.